2/contrib/jenkins/install-plugins.sh

```diff
@@ -200,6 +202,17 @@ function doDownload() {
         return 0
     fi
 
+    # Check if the plugin is cached and in correct version; if so, use it instead of downloading
+    # Some plugins do not follow the naming conventions and include the "-plugin" suffix; both cases need to be covered
+    for pluginFilename in "$plugin.jpi" "$plugin-plugin.jpi"; do
+        local cachedPlugin="$INCREMENTAL_BUILD_ARTIFACTS_DIR/plugins/$pluginFilename"
+        if test -f "$cachedPlugin" && [[ $(get_plugin_version "$cachedPlugin") == "$version" ]]; then
+            echo "Copying plugin from a cache created by s2i: $cachedPlugin"
+            cp "$cachedPlugin" "$jpi"
```

Comment:
i think this should be a move rather than a cp, so you don't end up w/ two copies of the plugin in the output image.

Reply:
even better, erase the entire /tmp/artifacts dir when done installing plugins, that way even plugins that didn't get reused are removed.

Reply:
Great point! For whatever reason I though s2i would clean that up automatically, but that's not the case. Fixing in now.

Reply:
I've updated the `assemble` script to delete the `tmp/artifacts/plugins` dir. It looks like that's the only place that needs this change - is that correct?